### PR TITLE
English translation of the documentation.

### PR DESCRIPTION
**Pull request by Hywan**
- [x] Abstract
- [x] Introduction
- [x] Global workflow
    - [x] Grammar
    - [x] Context
    - [x] Asserter
        - [x] Add functions
- [x] Language transformation
    - [x] Interpreter: language to object model
    - [x] Compiler: object model to PHP
    - [x] Disassembler: object model to language
- [x] Performances
    - [x] Serialize the object model
    - [x] Save and execute PHP code
- [x] Conclusion

### CODE REVIEW

**Comment by Hywan** [comment]
Please @hoaproject/hoackers, this translation is ready for a review!

See how to generate the documentation locally: https://github.com/hoaproject/Ruler/pull/55#issuecomment-78095587.

File: Documentation/En/Index.xyl [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+  allowing to simply execute business rules while being
+  <strong>efficient</strong> and very <strong>extensible</strong>.</p>
+
+  <h2 id="Table_of_contents">Table of contents</h2>
+
+  <tableofcontents id="main-toc" />
+
+  <h2 id="Introduction" for="main-toc">Introduction</h2>
+
+  <p>The business logic is very different of the Computer logic. “<q>All the
```
Comment (osaris) [inline Documentation/En/Index.xyl]: different than/from

File: Documentation/En/Index.xyl [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+  customers that have reached 100€ in one time receive a voucher of 10% on the
+  next purchase</q>”. This rule allows to access to certain parts of the program
+  if <strong>valids</strong>. However, it can <strong>change</strong> at any
+  moment. Most of time, in a team, it is not the role of the developer to
```
Comment (osaris) [inline Documentation/En/Index.xyl]: most of the time ?

File: Documentation/En/Index.xyl [hunk 3]
```diff
@@ -0,0 +1,710 @@
...
+  implement this rule. It will probably come from a <strong>business rules
+  repository</strong>, which have been written by other persons, either
+  <strong>manually</strong> or thanks to a <strong>third program</strong>.
```
Comment (osaris) [inline Documentation/En/Index.xyl]: third party program ?

File: Documentation/En/Index.xyl (hunk 1 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Fixed, thanks!

File: Documentation/En/Index.xyl (hunk 2 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Fixed, thanks!

File: Documentation/En/Index.xyl (hunk 3 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Is “third program” not enough?

File: Documentation/En/Index.xyl [hunk 4]
```diff
@@ -0,0 +1,710 @@
...
+      </tr>
+      <tr>
+        <td><code>is</code>, <code>in</code></td>
+        <td>membership operators</td>
+      </tr>
+    </tbody>
+  </table>
+
+  <p>Of course, these examples represent atomic parts of the grammar that we can
+  combined. Thus:
```
Comment (osaris) [inline Documentation/En/Index.xyl]: combine

File: Documentation/En/Index.xyl (hunk 3 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Fixed, thanks! (“third-party”)

File: Documentation/En/Index.xyl (hunk 4 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Fixed, thanks!

File: Documentation/En/Index.xyl [hunk 5]
```diff
@@ -0,0 +1,710 @@
...
+  <code>hoa://Library/Ruler/Grammar.pp</code> file</a>), we will get an AST: An
+  abstract syntax tree. For instance, for the rule <code>points > 30</code>, its
+  AST is:</p>
+  <pre><code class="language-shell">$ echo 'points > 30' | hoa compiler:pp hoa://Library/Ruler/Grammar.pp 0 --visitor dump
+>  #expression
+>  >  #operation
+>  >  >  token(identifier, points)
+>  >  >  token(identifier, >)
+>  >  >  token(integer, 30)</code></pre>
+  <p>In order this tree to be exploitable easier, it will be transformed into an
```
Comment (osaris) [inline Documentation/En/Index.xyl]: In order to be easily exploitable, this tree will ...

File: Documentation/En/Index.xyl [hunk 6]
```diff
@@ -0,0 +1,710 @@
...
+echo $disassembly->visit($model);
+
+/**
+ * Will output:
+ *     (points > 30)
+ */</code></pre>
+
+  <h2 id="Performances" for="main-toc">Performances</h2>
+
+  <p>Transforming a rule into an object model is not a low-cost operation if
```
Comment (osaris) [inline Documentation/En/Index.xyl]: not english specific but the fact that it's not a low-cost operation is not related to the fact that it's applied thousand times per minutes

File: Documentation/En/Index.xyl [hunk 7]
```diff
@@ -0,0 +1,710 @@
...
+ */</code></pre>
+  <p>This way, the rule is transformed into an object model <strong>only
+  once</strong>!</p>
+
+  <h3 id="Save_and_execute_PHP_code" for="main-toc">Save and execute PHP
+  code</h3>
+
+  <p>Another way to avoid the transformation of a rule into its object model is
+  to save the PHP code allowing to build the object model thanks to the
+  <code>Hoa\Ruler\Visitor\Compiler</code> class. Once this PHP code stored and
```
Comment (osaris) [inline Documentation/En/Index.xyl]: is stored ?

**Comment by osaris** [comment]
Good job :)

File: Documentation/En/Index.xyl (hunk 5 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: Fixed, thanks!

File: Documentation/En/Index.xyl (hunk 6 above)
Comment (Hywan) [inline Documentation/En/Index.xyl]: You're right, but the meaning is: “The cost becomes significant if we apply this operation several times”. Should we reformulate?

Comment (osaris) [inline Documentation/En/Index.xyl]: yes we should reformulate imo

**Comment by stephpy** [comment]
:+1:

**Comment by Hywan** [comment]
:heart:

**Comment by Hywan** [comment]
Website is updated: http://hoa-project.net/Literature/Hack/Ruler.html.

**Pull request merged**